judge/judgedaemon.main.php

```diff
@@ -92,9 +92,14 @@ function close_curl_handles(): void
  * $data is the urlencoded data passed as GET or POST parameters.
  * When $failonerror is set to false, any error will be turned into a
  * warning and null is returned.
+ * This function retries request on transient network errors.
+ * To deal with the transient errors while avoiding overloads,
+ * this function uses exponential backoff algorithm.
+ * Every error except HTTP 401, 500 is considered transient.
  */
 $lastrequest = '';
-function request(string $url, string $verb = 'GET', $data = '', bool $failonerror = true)
+function request(
+    string $url, string $verb = 'GET', $data = '', bool $failonerror = true, int $jitter = 200, $factor = 2, $steps = 3)
```

Comment:
I'm not sure it makes sense to add these as 3 extra optional parameters. In that case the parameter list becomes very long and unclear what each parameter means.

Reply:
Thanks. Moved them into etc/judgehost-config.php.